test/models/requests/CreateUpdateExpensesRequestSpec.scala

```diff
@@ -0,0 +1,235 @@
+           |		"otherAndCapitalAllowances": 21
+           |  },
+           |	"employmentExpensesData": {
+           |		"jobExpenses": 100,
+           |		"flatRateJobExpenses": 50,
+           |		"professionalSubscriptions": 140,
+           |		"otherAndCapitalAllowances": 210
+           |	}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+    "some prior expenses are provided " in {
+
+      val priorCustomerEmploymentExpenses = employmentExpenses.copy(
+        expenses = Some(defaultExpenses.copy(
+          businessTravelCosts = Some(15),
+          jobExpenses = Some(10),
+          flatRateJobExpenses = Some(5),
+          professionalSubscriptions = None,
+          hotelAndMealExpenses = Some(12),
+          otherAndCapitalAllowances = None,
+          vehicleExpenses = Some(25),
+          mileageAllowanceRelief = Some(30)
+        )))
+
+
+      val actualResult = defaultModel.toAmendAuditModel(taxYear, priorCustomerEmploymentExpenses)
+
+
+      val expectedJson = Json.parse(
+        s"""{
+           |	"taxYear": $taxYear,
+           |	"userType": "${user.affinityGroup.toLowerCase}",
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |  "priorEmploymentExpensesData": {
+           |		"jobExpenses": 10,
+           |		"flatRateJobExpenses": 5
+           |  },
+           |	"employmentExpensesData": {
+           |		"jobExpenses": 100,
+           |		"flatRateJobExpenses": 50,
+           |		"professionalSubscriptions": 140,
+           |		"otherAndCapitalAllowances": 210
+           |	}
+           |}
+           |""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+    "no prior expenses are provided" in {
+      val model = defaultModel.copy(
+        expenses = defaultExpenses.copy(
+          professionalSubscriptions = None,
+          otherAndCapitalAllowances = None
+        )
+      )
+
+      val priorExpenses = defaultPriorCustomerEmploymentExpenses.copy(
+        expenses = None
+      )
+
+      val actualResult = model.toAmendAuditModel(taxYear, priorExpenses)
+      val expectedJson = Json.parse(
+        s"""{
+           |	"taxYear": $taxYear,
+           |	"userType": "${user.affinityGroup.toLowerCase}",
+           |	"nino": "${user.nino}",
+           |	"mtditid": "${user.mtditid}",
+           |   "priorEmploymentExpensesData": {},
+           |	"employmentExpensesData": {
+           |		"jobExpenses": 100,
+           |		"flatRateJobExpenses": 50
+           |	}
+           |}""".stripMargin)
+      Json.toJson(actualResult) shouldBe expectedJson
+    }
+    "some new expenses are provided" in {
+      val model = defaultModel.copy(
+        expenses = defaultExpenses.copy(
+          professionalSubscriptions = None,
+          otherAndCapitalAllowances = None
+        )
+      )
+
+      val actualResult = model.toAmendAuditModel(taxYear, defaultPriorCustomerEmploymentExpenses)
```

Comment:
I think you added a test with no prior which was worthwhile but the one I had in mind was where there are no current expenses so we know all of them will be excluded from the audit if not present
Could you add this one too?

e.g. to produce this:
```
s"""{
           |	"taxYear": $taxYear,
           |	"userType": "${user.affinityGroup.toLowerCase}",
           |	"nino": "${user.nino}",
           |	"mtditid": "${user.mtditid}",
           |   "priorEmploymentExpensesData": {
           |		"jobExpenses": 10,
           |		"flatRateJobExpenses": 5,
           |		"professionalSubscriptions": 14,
           |		"otherAndCapitalAllowances": 21
           |  },
           |	"employmentExpensesData": {}
           |}""".stripMargin)
```